assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+    Player(element,options){
+        this.player_status = -1;
+        this.error_code = 0;
+        this.volume = 100;
+        this.eventobject = {ready:[],ended:[],error:[],ratechange:[],volumechange:[],waiting:[],timeupdate:[],loadedmetadata:[],play:[],seeking:[],seeked:[],playerresize:[],pause:[],statechange:[]};
+        var replace_elemnt;
+        if(element===undefined||element===null){
+            throw 'please set element id or HTMLElement';
+        }
+        else if(typeof element==='string'){
+            replace_elemnt = document.getElementById(element);
+        }
+        else{
+            replace_elemnt = element;
+        }
+        var iframe_src = '';
+        if(options.host!==undefined&&options.host!==""){
+            iframe_src = new URL(options.host).origin;
+        }
+        else{
+            iframe_src = 'https://vid.puffyan.us';//I set most hot instanse but this may need discuss or change ay to default instanse
+        }
+        this.target_origin = iframe_src.slice();
+        iframe_src += '/embed/';
+        if(typeof options.videoId==='string'&&options.videoId.length===11){
+            iframe_src += options.videoId;
+            this.videoId = options.videoId;
+        }
+        else{
+            this.error_code = 2;
+            this.event_executor('error');
+        }
+        var search_params = new URLSearchParams('');
+        search_params.append('widgetid',invidious_embed.widgetid);
+        this.widgetid = invidious_embed.widgetid;
+        invidious_embed.widgetid++;
+        search_params.append('origin',location.origin);
+        search_params.append('enablejsapi','1');
+        if(typeof options.playerVars==='object'){
+            this.option_playerVars = options.playerVars;
+            for(var x in options.playerVars){
+                if(typeof x==='string'&&typeof options.playerVars[x]==='string'){
+                    search_params.append(x,options.playerVars[x]);
+                }
+            }
+            if(options.playerVars.autoplay===undefined){
+                search_params.append('autoplay','0');
+            }
+        }
+        iframe_src += "?" + search_params.toString();
+        if(options.events!==undefined&&typeof options.events==='object'){
+            for(let x in options.events){
+                this.addEventListener(x,options.events[x]);
+            }
+        }
+        this.player_iframe = document.createElement("iframe");
+        this.loaded = false;
+        this.addEventListener('loadedmetadata',()=>{this.event_executor('ready');this.loaded=true});
+        this.addEventListener('loadedmetadata',()=>{this.setVolume(this.volume)});
+        this.player_iframe.src = iframe_src;
+        if(options.width!==undefined&&typeof options.width==='number'){
+            this.player_iframe.width = options.width;
+        }
+        else{
+            if(document.body.clientWidth < 640){
+                this.player_iframe.width = document.body.clientWidth;
```

Comment:
IMO, this may be done more accurately via setting CSS `max-width: 100%` of iframe element.